src/Security/InheritedPermissions.php

```diff
@@ -0,0 +1,632 @@
+        return array_fill_keys($deletable, true) + array_fill_keys($ids, false);
+    }
+
+    /**
+     * Check delete permission for a single record ID
+     *
+     * @param int $id
+     * @param Member $member
+     * @return bool
+     */
+    public function canDelete($id, Member $member = null)
+    {
+        // No ID: Check root permission
+        if (!$id) {
+            return $this->checkRootPermission(self::DELETE, $member);
+        }
+
+        // Regular canEdit logic is handled by canEditMultiple
+        $results = $this->canDeleteMultiple(
+            [ $id ],
+            $member
+        );
+
+        // Check if in result
+        return isset($results[$id]) ? $results[$id] : false;
+    }
+
+    /**
+     * Check edit permission for a single record ID
+     *
+     * @param int $id
+     * @param Member $member
+     * @return bool
+     */
+    public function canEdit($id, Member $member = null)
+    {
+        // No ID: Check root permission
+        if (!$id) {
+            return $this->checkRootPermission(self::EDIT, $member);
+        }
+
+        // Regular canEdit logic is handled by canEditMultiple
+        $results = $this->canEditMultiple(
+            [ $id ],
+            $member
+        );
+
+        // Check if in result
+        return isset($results[$id]) ? $results[$id] : false;
+    }
+
+    /**
+     * Check view permission for a single record ID
+     *
+     * @param int $id
+     * @param Member $member
+     * @return bool
+     */
+    public function canView($id, Member $member = null)
+    {
+        // No ID: Check root permission
+        if (!$id) {
+            return $this->checkRootPermission(self::VIEW, $member);
+        }
+
+        // Regular canView logic is handled by canViewMultiple
+        $results = $this->canViewMultiple(
+            [ $id ],
+            $member
+        );
+
+        // Check if in result
+        return isset($results[$id]) ? $results[$id] : false;
+    }
+
+    /**
+     * Get field to check for permission type for the given check
+     *
+     * @param string $type
+     * @return string
+     */
+    protected function getPermissionField($type)
+    {
+        switch ($type) {
+            case self::DELETE:
+                // Delete uses edit type - Drop through
+            case self::EDIT:
+                return 'CanEditType';
+            case self::VIEW:
+                return 'CanViewType';
+            default:
+                throw new InvalidArgumentException("Invalid argument type $type");
+        }
+    }
+
+    /**
+     * Get join table for type
+     *
+     * @param string $type
+     * @return string
+     */
+    protected function getJoinTable($type)
+    {
+        $table = DataObject::getSchema()->tableName($this->baseClass);
+        switch ($type) {
+            case self::DELETE:
+                // Delete uses edit type - Drop through
+            case self::EDIT:
+                return "{$table}_EditorGroups";
```

Comment:
This should cover 99% of use cases. As @unclecheese suggested I'm moving these fields to a standard extension, so if users want to use their own fields, they can subclass InheritedPermissions.